### ISSUE TITLE
[New rule] prefer_const_border_radius

### ISSUE THREAD

**Issue by grafovdenis** [dart-code-checker/dart-code-metrics#411]
**Please describe what the rule should do:**
[BorderRadius.circular constructor](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/border_radius.dart#L292-L295) calls `const BorderRadius.all` constructor under the hood.
This rule allows to replace `BorderRadius.circular(radius)` with `const BorderRadius.all(Radius.circular(radius))` if `radius` is a constant value.
With such replacement, you can pass default `BorderRadius` value into your widget constructor.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart
const _defaultRadius = BorderRadius.all(Radius.circular(8)); // Good
final _defaultFinalRadius = BorderRadius.circular(8); // Bad

class RoundedWidget extends StatelessWidget {
  final BorderRadius borderRadius;
  final Widget child;

  const RoundedWidget({
    Key? key,
    this.borderRadius = _defaultRadius,
    required this.child,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      borderRadius: borderRadius,
      child: child,
    );
  }
}

```

**Are you willing to submit a pull request to implement this rule?**
Most likely.
